### PR TITLE
Bug fixes, reorganise plugin directories, tidy

### LINKED ISSUE #41: Fix the problem of accessing static/class methods on plugins

### LINKED ISSUE #45: Plugins cannot contain the `tempor` substring in their name
## Description
A plugin with a name like `plugin_temporal_minmax_scaler.py`/`temporal_minmax_scaler` will break the PluginLoader.

### PR DESCRIPTION
**Pull request by DrShushen**
## Description
Closes #45 
Closes #41 
PR with several bug fixes, minor improvements and tidying, and reorganisation of plugins directory structure to be more hierarchical for clarity.

Main points:
* Fix #45 (45b259c42bdeef84e7b15cac0f7749ff1c41b0d6).
* Fix #41 (ff65c6a8575752f4176ae7111b88ca7f725d01c0).
* Speed up `Samples` validation by extending `pandera.DataType` to have `Union`, rather than using a manual `pandera` `Check` to iterate through possible `dtype`s (354b288c601c1141dbca9a789e0c2f40beaf3ea7).
* Significantly speed up the tests (1h+ to <30min on GH actions workflow) running time by:
    * Using `pytest-xdist` (cca6cce7b9f005c64597e5f5740e526d239c0dee, a882a373c6e8a09d537d92ec72d1d1f0b3ce4dde, 672134f5b57d473b43bd41f5309f4ca05de5d0c6).
    * Splitting out dataloaders/sets for the tests as fixtures, initialising them per test session, making testing datasets smaller (fc6879ed26f529343711f5e059861a0b5bca4f55, b121513f88bc7ce71bbb1e5e1ff44350219f2817, 9834e30064490a4d07bd1c1b45c3abd9955d7114).
* Added `extra` mark for more exhaustive tests, created a `test_full.yml` scheduled GH actions workflow for running the full set of tests (fbba0bf1cff3e7cf2a6aadf33a63b86233279bcf).
* Default to `Softmax` rather than `GumbelSoftmax` (7330f0771a08c43c9ccbea5b7742aead2d95cbb3).
* Add more `Dataloader`s, `clairvoyance2`'s PKPD, dummy (08b3f213e1278ac88f21cab6061a1cca3514f34f).
* Make `PluginLoader` refresh its list of plugins to make sure newly added plugins are reflected (da5db92b348f31859f75c3b48863746fe6a8b863).
* Ensure `pandas` `v2` compatibility (753481a604defb080ce4b29a634e544783608957, dbd6fd7cc66b60a5a4ef34a9d6a5371923819d00, 222b112d2e7cd00f1c4a94b23309aa15af7a25d2). 
* Reorganise plugins more hierarchically (a3404afa48515e3cd774ebfc643a2168323bba67, aa52f23d489b0a18296f1485bd1eccce8948b4f0, 5914886cd3075d008d8e08aa66a6246fd3994074, 57892ab5b1937b516afc33d22eaee56c737cbdb6).

## Affected Dependencies
* Added `packaging` to `install_requires` to make sure it's available to carry out dependency version checks (38d1aaff3b0086a8c008ef3d87aaaca56cb52030, 51d4ee786a3abb2fc67fbdb7e83e05c66e855f95).

## How has this been tested?
* New tests added as necessary:
    * [`test_importing`](https://github.com/vanderschaarlab/temporai/compare/main...DrShushen/bugfix-and-tidy?expand=1#diff-cfb684ac3c0e208a2863356397406937a175ff7bd1af5d5c388e65ea7cd63b6c)
    * [`test_pandera_utils`](https://github.com/vanderschaarlab/temporai/compare/main...DrShushen/bugfix-and-tidy?expand=1#diff-b69a029d15313df581be5eb5b9c065d72c82d65279d89fa38efa8db1dbaf7d21)
    * [`test_core_utils`](https://github.com/vanderschaarlab/temporai/compare/main...DrShushen/bugfix-and-tidy?expand=1#diff-0efab05715737b5552b3d787b6f09d3eeabef2a1d4111e4a6275f19c21c39751)
    * [`conftest.py`](https://github.com/vanderschaarlab/temporai/compare/main...DrShushen/bugfix-and-tidy?expand=1#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128) - new fixtures.
    * [`plugins/core/conftest.py`](https://github.com/vanderschaarlab/temporai/compare/main...DrShushen/bugfix-and-tidy?expand=1#diff-5cdeea3f1ae3675c3f7a62ef71a087b80db08ef6c712a260da3cdd3ba359dad0) - new fixtures.
    * [`plugins/preprocessing/scaling/conftest.py`](https://github.com/vanderschaarlab/temporai/compare/main...DrShushen/bugfix-and-tidy?expand=1#diff-7c5da394840126b8ee261f6724345c95fb71e7a7d4e04a85a06f02e5af76d3a3) - new fixtures.

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/vanderschaarlab/.github/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/vanderschaarlab/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [van der Schaar Lab Styleguide](https://github.com/vanderschaarlab/.github/blob/main/STYLEGUIDE.md)
- [X] I have labelled this PR with the relevant [Type labels](https://github.com/vanderschaarlab/.github/labels?q=Type%3A)
- [X] My changes are covered by tests

### CODE REVIEW

**Review by bcebere — APPROVED** [review]
LGTM
